src/rand_impls.rs

```diff
@@ -176,6 +207,94 @@ impl Rand for bool {
     }
 }
 
+impl Rand for Ordering {
+    #[inline]
+    fn rand<R: Rng>(rng: &mut R) -> Ordering {
+        match rng.gen_range(0, 3) {
+            0 => Ordering::Less,
+            1 => Ordering::Equal,
+            2 => Ordering::Greater,
+            _ => unreachable!(),
+        }
+    }
+}
+
+#[cfg(feature = "duration")]
+impl Rand for Duration {
+    #[inline]
+    fn rand<R: Rng>(rng: &mut R) -> Duration {
+        Duration::new(rng.gen(), rng.gen_range(0, 1_000_000_000))
```

Comment:
I would suggest using an exponential distribution here

Reply:
I consider `Duration` to be very similar to any other integer type, and so uniform probability makes the most sense to me.

Reply:
How often are you working with hundred million year durations?

Reply:
@clarcharr that's a null argument considering there are many things people might want to do with random integers (virtual dice, randomised network delays, random sorting, random selection from a list (e.g. a set of Unicode glyphs, Poisson distributions) and probably none of them involve uniform sampling from a fixed range not specific to this use-case.

I suggested exponential here because that, or sampling from a short arbitrary range (e.g. 0 to 1 minutes) are about the only things that make any sense whatsoever when you don't know the target use.